modules/hal_silabs/wiseconnect/CMakeLists.txt

```diff
@@ -156,9 +156,30 @@ if(CONFIG_BT_SILABS_SIWX91X)
 endif() # CONFIG_BT_SILABS_SIWX91X
 
 if(CONFIG_SILABS_SIWX91X_NWP)
+  dt_nodelabel(sram0_path NODELABEL "sram0")
+  dt_prop(sram0_reg PATH "${sram0_path}" PROPERTY "reg")
+
+  # sram0_reg is now a list of values (start address, size)
+  # Extract the second value (index 1: size in bytes)
+  list(GET sram0_reg 1 sram0_size_bytes)
+
+  # Convert bytes to kB (CMake integers are always decimal)
+  math(EXPR sram0_size_kb "${sram0_size_bytes} / 1024")
+
+  # Assign RAM_MEM_CONFIG based on the kB value
+  if(sram0_size_kb EQUAL 195)
+     set(RAM_MEM_CONFIG 1)
+  elseif(sram0_size_kb EQUAL 255)
+     set(RAM_MEM_CONFIG 2)
+  elseif(sram0_size_kb EQUAL 319)
+     set(RAM_MEM_CONFIG 3)
+  else()
+     message(FATAL_ERROR "Unsupported RAM size config: ${sram0_size_kb}")
+  endif()
+
   zephyr_compile_definitions(
```

Comment:
change `zephyr_compile_definitions` in this file to `zephyr_library_compile_definitions`